### DISCUSSION TITLE
Getting "You are not allowed to run sudo" error when trying to use sudo

### DISCUSSION THREAD

**Discussion by gr33nmonk3y** [microsoft/sudo#68] [Q&A]
I just enabled sudo finally in Windows 11 (version 24H2), and can not get it to work with a non-administrator local account. Im under the impression that I should be able to log into the machine with a normal user account, and from a normal terminal window use sudo to run administratively privileged commands and get prompted for the local administrator account credentials. Am I doing something wrong here?

**Comment by AvogatoWizardWhisker** [comment]
Did you enable from terminal or in Windows' Settings app?

Edit:  If you want to enable `sudo` from Windows Terminal, you need to run it as administrator, then run `sudo config --enable <configuration_option>`. For the configuration option, you can choose `forceNewWindow`, `disableInput` or `normal`.
Or, just open Settings app on Windows 11 `Settings > For Developers`, then chose what configuration you want.

**Reply by TurtleShroom** [reply]
On my W11 installation, there is no "Enable Sudo" button on the Windows Settings Developer Mode menu. However, I AM running the Administrative account credential AND the Power Shell Terminal is opened in an elevated prompt. What am I doing wrong?

**Comment by zadjii-msft** [comment]
Sudo for Windows doesn't currently support over-the-shoulder elevation (where the admin user is a different user than the current user). I'd suspect that we'll address that when we add support for #3

**Reply by gr33nmonk3y** [reply]
This sounds like my issue - ill test today to confirm, but thanks for the explanation.

**Reply by mobilejon** [reply]
So specifically are we saying the only person who can use Sudo are people with local admin already?

**Reply by AlphaSeb** [reply]
Wow, then this whole feature is pretty useless in any enterprise enviroment

**Reply by KSatC5** [reply]
This makes sudo on Windows completely pointless. The whole idea of sudo is to be logged in as a non-admin user, but to elevate for one task.

**Reply by sean729** [reply]
When I read Windows 11 (24H2) supports sudo I thought that meant actually sudo, not a binary called sudo.exe that resembles sudo in name only. Disappointing.

**Reply by roke-julian-lockhart** [reply]
@sean729, I'd rather it be available now without *every* possible feature of `sudo`, and be *an* improvement, than have them wait until it's perfect before release. What's the point in belittling the developers because you didn't do your research? I find it damn useful.

**Comment by Sellinfish** [comment]
I got this error too, I try to enable group policy setting item: User Account Control: Run all Administrators in Admin Approval Mode.
Check the UAC setting can help you when getting "You are not allowed to run sudo" error.

**Comment by mobilejon** [comment]
That’s only if you don’t have local admin.

**Reply by AlphaSeb** [reply]
False. We have a local admin, protected by LAPS and personalized Admin Accounts assigned to our Entra ID Device Administrator Role. We also receive the same error message. It only seems to work if the same user happens to be also in the Local Admin Group, which is stupid and nobody in a business context would work like that.

If I type "sudo xyz" I would expect an UAC prompt which I would confirm with my credentials, not an error message.

**Reply by Fifteen15Studios** [reply]
> If I type "sudo xyz" I would expect an UAC prompt which I would confirm with my credentials, not an error message.

This is what I would expect, too. The fact that they released this, and announced it as a "feature", is laughable. The current implementation is literally useless.

**Comment by mobilejon** [comment]
So you’re expecting it to prompt for credentials?

I know for me it works if the user is part of local admin group on the PC

**Reply by AlphaSeb** [reply]
Sure, it should give an UAC prompt. If you're in the local admin group you receive a "Yes" "No" prompt. In most enterprise enviroments you'd see a prompt like that.

![image](https://github.com/user-attachments/assets/0b80fe26-8d43-4d41-9005-bad38fb5aafd)

**Comment by mobilejon** [comment]
Yeah I think Microsoft’s spirit of sudo is different from Linux. It seems
clear in 24H2 it’s about protecting admin accounts with this and just in
time admin coming at Ignite.

I think it’s good feedback but not sure that’s where they’re going with it

**Comment by mobilejon** [comment]
As an aside, other user is on the roadmap just not sure when they’re
planning to release
